src/Tribe/Admin/Pages.php

```diff
@@ -0,0 +1,274 @@
+	 *   @type string      path         Path for this page, full path in app context; ex /analytics/report
+	 *   @type string      capability   Capability needed to access the page.
+	 *   @type string      icon         Icon. Dashicons helper class, base64-encoded SVG, or 'none'.
+	 *   @type int         position     Menu item position.
+	 *   @type int         order        Navigation item order.
+	 * }
+	 *
+	 * @return string $page The resulting page's hook_suffix.
+	 *
+	 */
+	public function register_page( $options ) {
+		$defaults = [
+			'id'         => null,
+			'parent'     => null,
+			'title'      => '',
+			'capability' => self::get_capability(),
+			'path'       => '',
+			'icon'       => '',
+			'position'   => null,
+			'callback'   => [ __CLASS__, 'render_page' ],
+		];
+
+		$options = wp_parse_args( $options, $defaults );
+
+		if ( is_null( $options['parent'] ) ) {
+			$page = add_menu_page(
+				$options['title'],
+				$options['title'],
+				$options['capability'],
+				$options['path'],
+				$options['callback'],
+				$options['icon'],
+				$options['position']
+			);
+		} else {
+			$page = add_submenu_page(
+				$options['parent'],
+				$options['title'],
+				$options['title'],
+				$options['capability'],
+				$options['path'],
+				$options['callback'],
+			);
+		}
+
+		$this->connect_page( $options );
+
+		return $page;
+	}
+
+	/**
+	 * Get the current page.
+	 *
+	 * @since TBD
+	 *
+	 * @return array|boolean Current page or false if not registered with this controller.
+	 */
+	public function get_current_page() {
+		if ( is_null( $this->current_page ) ) {
+			$this->determine_current_page();
+		}
+
+		return $this->current_page;
+	}
+
+	/**
+	 * Determine the current page.
+	 *
+	 * @since TBD
+	 *
+	 * @return string|boolean Current page or false if not registered with this controller.
+	 */
+	public function determine_current_page() {
+		$current_screen = get_current_screen();
+
+		if ( is_null( $current_screen ) ) {
+			$this->current_page = isset( $_GET['page'] ) ? $_GET['page'] : null;
+			return $this->current_page;
+		}
+
+		$this->current_page = $current_screen->id;
+
+		return $this->current_page;
+	}
+
+	/**
+	 * Connect an existing page to wp-admin.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $options {
+	 *   Array describing the page.
+	 *
+	 *   @type string       id           Id to reference the page.
+	 *   @type string|array title        Page title. Used in menus and breadcrumbs.
+	 *   @type string|null  parent       Parent ID. Null for new top level page.
+	 *   @type string       path         Path for this page. E.g. admin.php?page=wc-settings&tab=checkout
+	 *   @type string       capability   Capability needed to access the page.
+	 *   @type string       icon         Icon. Dashicons helper class, base64-encoded SVG, or 'none'.
+	 *   @type int          position     Menu item position.
+	 * }
+	 */
+	public function connect_page( $options ) {
```

Comment:
Either type-hint `$options` as `array`, or handle the case where `$options` is not an array.